### DISCUSSION TITLE
An inconvenient situation when porting with a custom triplet

### DISCUSSION THREAD

**Discussion by playgithub** [microsoft/vcpkg#25501] [General]
e.g. some lib needs cpp17, so in my own branch of vcpkg
* Add `vcpkg\triplets\community\x64-windows-cpp17.cmake`
* Add environment variables
    * VCPKG_DEFAULT_HOST_TRIPLET=x64-windows-cpp17
    * VCPKG_DEFAULT_TRIPLET=x64-windows-cpp17

But a port branch should fork from master, which doesn't have `vcpkg\triplets\community\x64-windows-cpp17.cmake`.
The normal solution is to delete environment variables
* VCPKG_DEFAULT_HOST_TRIPLET
* VCPKG_DEFAULT_TRIPLET
when porting, add add them back when not porting, it is very inconvenient.

Any better solution?

**Comment by playgithub** [comment]
A dedicated Windows account for vcpkg is a way.

**Comment by Neumann-A** [comment]
Use overlay-triplets via [VCPKG_OVERLAY_TRIPLETS](https://github.com/microsoft/vcpkg/blob/master/docs/users/config-environment.md#vcpkg_overlay_triplets)

**Reply by Neumann-A** [reply]
also consider moving your triplet out of the vcpkg tree.

**Reply by playgithub** [reply]
Then env var `VCPKG_OVERLAY_TRIPLETS` needs to be modified manually instead of `VCPKG_DEFAULT_TRIPLET`, it seems no more convenient.
